service-workers/service-worker/iframe-controller.https.html

```diff
@@ -0,0 +1,95 @@
+<!DOCTYPE html>
+<title>Service worker inheritance</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/test-helpers.sub.js"></script>
+<script>
+
+promise_test(function(t) {
+    var script = 'iframe-controller-worker.https.js';
+    var scope1 = 'iframe-controller.https.html';
+    var scope2 = 'resources/blank.html';
+    var script1 = script + '?for_scope1';
+    var script2 = script + '?for_scope2';
+    var registration1, registration2;
+    var parent_controller, child_controller;
+    return service_worker_unregister_and_register(t, script1, scope1)
+      .then(function(r) {
+          registration1 = r;
+          return wait_for_state(t, registration1.installing, 'activated');
+        })
+      .then(function() {
+          return service_worker_unregister_and_register(t, script2, scope2);
+        })
+      .then(function(r) {
+          registration2 = r;
+          return wait_for_state(t, registration2.installing, 'activated');
+        })
+      .then(function() {
+          // This client's active service worker is set to registration1's
+          // active worker by clients.claim() during the activation.
+          parent_controller = navigator.serviceWorker.controller;
+          return with_iframe('');
+        })
+      .then(function(f) {
+          child_controller = f.contentWindow.navigator.serviceWorker.controller;
+          assert_equals(child_controller.scriptURL, parent_controller.scriptURL,
+                        'about:blank Document should inherit parent\'s SW.');
+        })
+      .then(function() {
+          return with_srcdoc_iframe('<p>Some HTML</p>');
```

Comment:
It would be nice if the various tests (about:blank, srcdoc, etc) would be actually separate promise_tests, rather than just assertions all in the same test. That way the test results would much clearer show which features work/don't work, rather than only reporting the first failing result. Not sure how tricky that is going to be with the various service worker registrations, but it shouldn't be too bad I think.

One way might be to do something like (but then with proper syntax etc):
```js
let registered_workers = service_worker_unregister_and_register(...).then(...).then(...);

promise_test(t => {
  registered_workers.then(() => with_iframe(''))
    .then(f => {
       assert(...)
     });
  }, 'about:blank document');

promise_test(t => {....}, 'srcdoc iframe');

// etc
```